### PR TITLE
Add new example/demo: three-dimensional fractals

### PR DESCRIPTION
**Pull request by rkharsan**
Added a new example/demo with three fractals. These are built using `repeat_sources`.

https://user-images.githubusercontent.com/72246047/162490498-148dad0b-a852-48ea-84ba-775a194c187f.mp4

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Hi @rkharsan,

Really nice tutorial. Thank you for this. It is almost ready to go. 

Can you check and update my comments?

Thank you. 

ps: I really like this tutorial

File: docs/examples/viz_fractals.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+    # primitive is scaled down depending on level
+    vertices /= 2 ** (N - 1)
+
+    # compute some pretty colors
+    bounds_min, bounds_max = np.min(centers, axis=0), np.max(centers, axis=0)
+    colors = (centers - bounds_min) / (bounds_max - bounds_min)
+
+    return utils.repeat_sources(centers=centers, colors=colors,
+                                vertices=vertices, faces=faces)
```
[the remark this answers is not in the document]
Comment (rkharsan) [reply docs/examples/viz_fractals.py:89]: I must have overlooked this. Thanks for the suggestion. I'll replace it.

**Comment by skoudoro** [comment]
I merged your PR concerning `prim_tetrahedron()`. So feel free to simplify this PR or let me know if you are done with it. 

Thanks again

**Comment by rkharsan** [comment]
I have simplified the code and rebased my branch. Ready to merge.

**Comment by skoudoro** [comment]
Thank you for this @rkharsan. merging

**Pull request merged**